src/main.ts

```diff
@@ -202,18 +201,18 @@ class Benchmark extends utils.Adapter {
 			};
 
 			// check all requested monitoring
-			for (const instance of Object.keys(this.requestedMonitoring)) {
+			for (const [instance, result] of Object.entries(this.requestedMonitoring)) {
```

Comment:
In TS it is generally not a good idea to excessively use `variable[property]` to access the same property over and over. Besides making the code harder to read, there are some limitations in TypeScript that you can prevent by assigning the value you access to a const (or use `Object.entries` like in this case)